src/v/cluster/rm_stm.cc

```diff
@@ -1632,15 +1633,26 @@ void rm_stm::maybe_rearm_autoabort_timer(time_point_type deadline) {
 }
 
 ss::future<tx::errc> rm_stm::abort_all_txes() {
+    static constexpr uint max_concurrency = 5u;
     if (!co_await sync(_sync_timeout())) {
         co_return tx::errc::stale;
     }
 
     tx::errc last_err = tx::errc::none;
 
+    // snap the intrusive list produced_ids before yielding the cpu
+    chunked_vector<model::producer_identity> producer_ids_to_expire;
```

Comment:
maybe we should just change the type of `_active_tx_producers` container @bharathv ?

Reply:
umm would that help? Any container is prone to iterator invalidation?

Reply:
sgtm!